pkg/server/plugin/notifier/k8sbundle/k8sbundle_watcher.go

```diff
@@ -0,0 +1,112 @@
+package k8sbundle
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type bundleWatcher struct {
+	p        *Plugin
+	clients  []kubeClient
+	watchers []watch.Interface
+	config   *pluginConfig
+
+	hooks struct {
+		watch func(ctx context.Context) error
+	}
+}
+
+// newBundleWatcher creates a new watcher for newly created objects
+func newBundleWatcher(ctx context.Context, p *Plugin, config *pluginConfig) (*bundleWatcher, error) {
+	clients, err := p.hooks.newKubeClient(config)
+	if err != nil {
+		return nil, err
+	}
+	watchers, clients, err := newWatchers(ctx, clients, config)
+	if err != nil {
+		return nil, err
+	}
+
+	watcher := &bundleWatcher{
+		p:        p,
+		clients:  clients,
+		config:   config,
+		watchers: watchers,
+	}
+	watcher.hooks.watch = watcher.watch
+	return watcher, nil
+}
+
+// Watch calls the hook to watch for new objects
+func (b *bundleWatcher) Watch(ctx context.Context) error {
+	return b.hooks.watch(ctx)
+}
+
+// watch watches for new objects that are created with the proper selector and updates the CA Bundle
+func (b *bundleWatcher) watch(ctx context.Context) error {
+	selectCase := newSelectCase(ctx, b.watchers)
+	for {
+		chosen, recv, _ := reflect.Select(selectCase)
+		if chosen < len(b.clients) {
+			if err := b.watchEvent(ctx, b.clients[chosen], recv.Interface().(watch.Event)); err != nil {
+				return k8sErr.New("handling watch event: %v", err)
```

Comment:
I ran into a boostrapping issue when testing today, gettting this error here: 
```
time="2021-03-09T23:14:44Z" level=error msg="Unable to watch" error="k8s-bundle: handling watch event: rpc error: code = FailedPrecondition desc = IdentityProvider host service has not been initialized" subsystem_name=built-in_plugin.k8sbundle
```
It happens when  a webhook is deployed prior to the spire server being deployed. The watcher gets set up before the IdentityProvider gets setup, and we get the add event from the webhook. Is there a place we can hook into to set up the watcher after IdentityProvider is initialized?

Reply:
Giving it some thought, option 1 is probably easiest considering all of the mechanics in Configure() around watcher reconfiguration, etc.

Reply:
Ok i made this change. I also made the watcher not exit on error to be more consistent with the way updateBundles behaves. LMK if this ok or if you prefer it to exit on error.